src/QsCompiler/Tests.Compiler/TypeParameterTests.fs

```diff
@@ -501,25 +501,39 @@ type TypeParameterTests () =
         AssertCombinedResolutionFailure expected given
 
     [<Fact>]
-    [<Trait("Category","Type resolution")>]
-    member this.``Nested Self Resolution`` () =
+    [<Trait("Category","Type Resolution")>]
+    member this.``Constricting Nested Self Resolution`` () =
+        let given = [|
+            ResolutionFromParam [
+                (FooA, [FooA |> TypeParameter; Int] |> MakeTupleType)
+            ]
+        |]
+        let expected = ResolutionFromParam [
+            (FooA, [FooA |> TypeParameter; Int] |> MakeTupleType)
+        ]
+
+        AssertCombinedResolutionFailure expected given
+
+    [<Fact>]
+    [<Trait("Category","Type Resolution")>]
+    member this.``Constricting Indirect Nested Self Resolution`` () =
```

Comment:
It would be good to also add the tests for arrays.

Reply:
I've added tests for nested self references in arrays in addition to the tests for nested self references in tuples.